pkg/controller/operconfig/operconfig_controller.go

```diff
@@ -208,15 +208,23 @@ func (r *ReconcileOperConfig) Reconcile(request reconcile.Request) (reconcile.Re
 	// Set up the Pod reconciler before we start creating DaemonSets/Deployments
 	daemonSets := []types.NamespacedName{}
 	deployments := []types.NamespacedName{}
+	relatedObjects := []configv1.ObjectReference{}
 	for _, obj := range objs {
 		if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "DaemonSet" {
 			daemonSets = append(daemonSets, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		} else if obj.GetAPIVersion() == "apps/v1" && obj.GetKind() == "Deployment" {
 			deployments = append(deployments, types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()})
 		}
+		relatedObjects = append(relatedObjects, configv1.ObjectReference{
+			Group:     obj.GetObjectKind().GroupVersionKind().Group,
+			Resource:  obj.GetKind(),
```

Comment:
This isn't right - the object kind is "Namespace", whereas the API resource is "namespaces". You need to actually look up the conversion between the Kind and the Resource using the the RESTMapper.

It's a bit awkward, yes. It looks like a few other operators do it. See if they hard-code the mapping or do something cleverer.